File: src/test/javascript/stepDefinitions.js
var assert = require('assert'),
    toRoman = require('./toRoman');

module.exports = function () {
    var result;

    this.When(/^I convert (.*) to Roman$/, function (arabic) {
        result = toRoman(arabic);
        console.log("result for arabic", result, arabic)
    });

    this.Then(/^I should get (.*)$/, function (roman) {
        assert.equal(result, roman);
    });
}

File: src/test/scala/StepDefinitions.scala
import cucumber.api.scala.{ScalaDsl, EN}
import ToRoman.toRoman

class StepDefinitions extends ScalaDsl with EN {

  var result: String = null

  When( """I convert (\d+) to Roman""") { (arabic: Integer) =>
    result = toRoman(arabic)
  }

  Then( """I should get (.*)""") { (expectedResult: String) =>
    assert(result == expectedResult)
  }

}
